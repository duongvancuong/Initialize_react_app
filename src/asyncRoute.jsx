import React from 'react';
import Loadable from 'react-loadable';
import Loading from './components/common/Loading';
import store, { injectAsyncSagas, injectAsyncReducer } from './store';

const asyncRoute = getRoute => Loadable({
  loader: getRoute,
  loading() {
    return <Loading />;
  },
  render(route, props) {
    const { Component, sagas, reducer } = route.default;
    if (sagas) {
      injectAsyncSagas({
        name: sagas.name,
        sagas: sagas.sagas,
        store,
      });
    }
    if (reducer) {
      injectAsyncReducer({
        name: reducer.name,
        asyncReducer: reducer.reducer,
        store,
      });
    }
    return <Component {...props} />;
  },
});

export default asyncRoute;
