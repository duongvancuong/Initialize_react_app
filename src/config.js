const configure = {
  baseURL: process.env.REACT_APP_API_BASE_URL,
  baseURLCoinMaker: process.env.COIN_MAKE_CAP_API_BASE_URL,
};

export default configure;
