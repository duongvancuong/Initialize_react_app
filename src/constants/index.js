import keyMirror from 'fbjs/lib/keyMirror';

export const ActionsTypes = keyMirror({
  AUTHENTICATE_USER: undefined,
  AUTHENTICATE_USER_SUCCESS: undefined,
  AUTHENTICATE_USER_ERROR: undefined,
  LOADING_USER_DATA: undefined,
  LOGOUT_USER: undefined,
  LOGOUT_USER_SUCCESS: undefined,
  CLEAN_ERROR_LOGIN: undefined,
  CLEAN_ERROR_LOGIN_SUCCESS: undefined,
  HANDLE_EXCEPTION_USER_ERROR: undefined,
});
