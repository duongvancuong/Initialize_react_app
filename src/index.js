import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import './lib';
import './styles/base/_globalStyle';
import './styles/base/_reset';
import store from './store';
import App from './components/layout/App';
import { theme } from './styles/abstracts/theme';
import registerServiceWorker from './registerServiceWorker';

const rootEl = document.getElementById('root');

render(
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      <App />
    </ThemeProvider>
  </Provider>,
  rootEl
);
registerServiceWorker();

if (module.hot && process.env.NODE_ENV !== 'production') {
  module.hot.accept('./components/layout/App', () => {
    render(<Provider store={store}><App /></Provider>, rootEl);
  });
}
