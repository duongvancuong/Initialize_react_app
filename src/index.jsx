import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import './lib';
import './styles/base/_globalStyle';
import './styles/base/_reset';
import store from './store';
import App from './components/layout/App';
import registerServiceWorker from './registerServiceWorker';
import SiteThemeProvider from './context/SiteThemeContext';

const rootEl = document.getElementById('root');

render(
  <Provider store={store}>
    <SiteThemeProvider>
      <App />
    </SiteThemeProvider>
  </Provider>,
  rootEl,
);
registerServiceWorker();

if (module.hot && process.env.NODE_ENV !== 'production') {
  module.hot.accept('./components/layout/App', () => {
    render(<Provider store={store}><App /></Provider>, rootEl);
  });
}
