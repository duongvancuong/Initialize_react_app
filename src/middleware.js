import createSagaMiddleware from 'redux-saga';
import { routerMiddleware } from 'connected-react-router';
import history from './modules/history';

export const sagaMiddleware = createSagaMiddleware();

const middleware = [
  routerMiddleware(history),
  sagaMiddleware,
];

export default middleware;
