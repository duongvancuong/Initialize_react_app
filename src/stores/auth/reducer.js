import { combineActions, handleActions } from 'redux-actions';

import {
  authenticateUserSuccess,
  authenticateUserError,
  loadingUserData,
  logoutUserSuccess,
  cleanErrorLoginSuccess,
  handleExceptionUserError,
} from './actions';

const initialState = {
  isAuthenticated: false,
  refreshToken: '',
  token: '',
  expiredAt: '',
  isLoading: false,
  error: '',
};

const auth = handleActions({
  [combineActions(
    authenticateUserSuccess,
    authenticateUserError,
    loadingUserData,
    logoutUserSuccess,
    cleanErrorLoginSuccess,
    handleExceptionUserError,
  )]: (state, action) => ({
    ...state, ...action.payload,
  }),
},
initialState);

export default auth;
