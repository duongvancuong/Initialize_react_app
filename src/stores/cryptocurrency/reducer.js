/* eslint-disable */
import { combineActions, handleActions } from 'redux-actions';

import {
  fetchCryptocurrencyAction,
  fetchCryptocurrencyErrorAction,
  fetchCryptocurrencySuccAction,
  loadingAction,
} from './actions';

const initialState = {
  status: {},
  data: [],
  isLoading: false,
  error: '',
}

const cryptocurrency = handleActions({
  [combineActions(fetchCryptocurrencyAction, fetchCryptocurrencyErrorAction, loadingAction, fetchCryptocurrencySuccAction)]: (state, action) => ({
    ...state, ...action.payload
  })},
  initialState
);

export default cryptocurrency;
