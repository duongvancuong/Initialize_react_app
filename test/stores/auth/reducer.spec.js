import reducer from '../../../src/stores/auth/reducer';
import {
  authenticateUser,
  authenticateUserSuccess,
  authenticateUserError,
  loadingUserData,
  logoutUser,
  logoutUserSuccess,
  cleanErrorLogin,
  cleanErrorLoginSuccess,
  handleExceptionUserError,
} from '../../../src/stores/auth/actions';
import { ActionsTypes } from '../../../src/constants';


describe('Auth', () => {
  let auth = reducer(undefined, {});

  it('should return init state', ()=> {
    expect(reducer(auth, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionsTypes.AUTHENTICATE_USER}`, () => {
    const values = {email: 'test@gmail.com', password: 'password123'}
    let auth = reducer(auth, authenticateUser(values));
    expect(auth).toMatchSnapshot();
  });

  it(`should handle ${ActionsTypes.AUTHENTICATE_USER_SUCCESS}`, () => {
    const values = {
      error: "",
      expiredAt: "123123123",
      isAuthenticated: true,
      isLoading: false,
      refreshToken: "123123123ASFCXD",
      token: "123123123ASFCXD",
    };
    let auth = reducer(auth, authenticateUserSuccess(values));
    expect(auth).toMatchSnapshot();
  });

  it(`should handle ${ActionsTypes.AUTHENTICATE_USER_ERROR}`, () => {
    const values = {
      error: "602",
      expiredAt: "",
      isAuthenticated: false,
      isLoading: false,
      refreshToken: "",
      token: "",
    };
    let auth = reducer(auth, authenticateUserError(values));
    expect(auth).toMatchSnapshot();
  });
})
