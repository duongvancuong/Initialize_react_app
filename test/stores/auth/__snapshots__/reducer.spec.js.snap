// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should handle AUTHENTICATE_USER 1`] = `
Object {
  "error": "",
  "expiredAt": "",
  "isAuthenticated": false,
  "isLoading": false,
  "refreshToken": "",
  "token": "",
}
`;

exports[`Auth should handle AUTHENTICATE_USER_ERROR 1`] = `
Object {
  "error": "602",
  "expiredAt": "",
  "isAuthenticated": false,
  "isLoading": false,
  "refreshToken": "",
  "token": "",
}
`;

exports[`Auth should handle AUTHENTICATE_USER_SUCCESS 1`] = `
Object {
  "error": "",
  "expiredAt": "123123123",
  "isAuthenticated": true,
  "isLoading": false,
  "refreshToken": "123123123ASFCXD",
  "token": "123123123ASFCXD",
}
`;

exports[`Auth should return init state 1`] = `
Object {
  "error": "",
  "expiredAt": "",
  "isAuthenticated": false,
  "isLoading": false,
  "refreshToken": "",
  "token": "",
}
`;
